homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -33,14 +33,29 @@ async def async_setup_entry(
     name = config_entry.data[CONF_NAME]
     pvpc_data_handler = PVPCData(
         tariff=config_entry.data[ATTR_TARIFF],
+        power=config_entry.data[ATTR_POWER],
+        power_valley=config_entry.data[ATTR_POWER_P3],
         local_timezone=hass.config.time_zone,
         websession=async_get_clientsession(hass),
         logger=_LOGGER,
         timeout=_DEFAULT_TIMEOUT,
     )
-    async_add_entities(
-        [ElecPriceSensor(name, config_entry.unique_id, pvpc_data_handler)], False
-    )
+    async_add_entities([ElecPriceSensor(name, config_entry, pvpc_data_handler)], False)
+
+
+async def options_update_listener(
+    hass: HomeAssistant, entry: config_entries.ConfigEntry
+):
```

Comment:
Missing return value typing. Please type the whole signature if adding type annotations.